provider/email.go

```diff
@@ -0,0 +1,91 @@
+package provider
+
+import (
+	"fmt"
+	"github.com/abahmed/kwatch/event"
+	"github.com/sirupsen/logrus"
+	gomail "gopkg.in/mail.v2"
+	"math"
+	"strings"
+)
+
+type email struct {
+	from     string
+	password string
+	host     string
+	port     int
+	to       string
+}
+
+// NewEmail returns new email instance
+func NewEmail(from string, password string, host string, port int, to string) Provider {
+	if len(from) == 0 {
+		logrus.Warnf("initializing email with an empty from")
+	} else if len(password) == 0 {
```

Comment:
I think we don't need else here to detect all missing values

Reply:
updated